src/treq/test/test_content.py

```diff
@@ -173,3 +184,21 @@ def test_text_content_default_encoding_no_header(self):
         self.protocol.connectionLost(Failure(ResponseDone()))
 
         self.assertEqual(self.successResultOf(d), u'\xa1')
+
+    def test_text_content_unicode_headers(self):
+        """
+        Header parsing is robust against unicode header names and values.
+        """
+        self.response.headers = Headers({
+            b'Content-Type': [
```

Comment:
Reading this test, I'm not clear how this can happen in real life.

Due to an odd quirk of the documentation generator, `Headers.__init__` doesn't get documented properly, but the relevant `@ivar` entry for the constructor argument is here:

https://twistedmatrix.com/documents/17.1.0/api/twisted.web.http_headers.Headers._rawHeaders.html

It's not valid to pass `unicode` (or 3-`str`) objects in to `Headers` at all; the internal representation is `bytes`.  So: how and where does this happen?